packages/react-components/react-nav-preview/src/components/NavItem/useNavItem.ts

```diff
@@ -53,6 +53,11 @@ export const useNavItem_unstable = (props: NavItemProps, ref: React.Ref<HTMLAnch
       }),
       { elementType: 'a' },
     ),
+    icon: slot.optional(icon, {
+      renderByDefault: false,
+      elementType: 'span',
+      defaultProps: { children: icon },
```

Comment:
This is probably going to cause errors because you are passing a slot as children. Maybe it was left from the previous implementation.
```suggestion
```

Reply:
Yeah, I was drafting off the Tabs implementation. This may just be crusty code.

Reply:
Ended up sticking with flexbox. But used similair positioning strategies, including logical positioning instead of negative left margine.